hayha/cloudformation_types.py

```diff
@@ -21,33 +21,151 @@ def __init__(self, outgoing_connection=None, incoming_connection=None, connectio
+            subject to protection/authentication by resource “A”. In these cases, the
+            minimum security level required to pass protection/authentication by “A” shall
+            be added onto the minimum security level required to reach “B”. Their sum
+            becomes the minimum security level required to reach “C”.
+
+        `entrance_protection_by`:
+            Description: A key-value property in the parent resource type(“A”) that
+            specifies that all incoming connections to “A” are protected by the
+            specified resource type (“B”), using “K”, which is a CloudFormation
+            property/key name in “A”. In the context of the Hayha tool/files, “K” refers
+            to its respective CloudFormation value, which is a reference pointer to “B”.
+            If any resource type (“C”) is configured to have an ‘outgoing_connection’
+            to “A” or if “A” is configured to have an ‘incoming_connection’ from “C” ,
+            that connection will be subject to protection/authentication by resource “B”.
+            In these cases, the minimum security level required to pass protection/authentication
+            by “B” shall be added onto the minimum security level required to reach
+            “C”. Their sum becomes the minimum security level required to reach “A”.
+
+        ‘exit_protection_by’:
+            Description: A key-value property in the parent resource type(“A”) that
+            specifies that all outgoing connections from “A” are protected by the
+            specified resource type (“B”), using “K”, which is a CloudFormation property/key name
+            in “A”. In the context of the Hayha tool/files, “K” refers to its respective
+            CloudFormation value, which is a reference pointer to “B”. If any resource type (“C”)
+            is configured to have an ‘incoming_connection’ from “A” or if “A” is
+            configured to have an ‘outgoing_connection’ to “C”, that connection will
+            be subject to protection/authentication by resource “B”. In these cases, the
+            minimum security level required to pass protection/authentication by “B” shall
+            be added onto the minimum security level required to reach “A”. Their
+            sum becomes the minimum security level required to reach “C”.
+
+        `connection_to_protect`:
+            Description: A key-value property in the parent resource type (“A”) that
+            specifies that “A” protects a specific connection with a specified resource
+            type (“B”) and another resource type (“C”), using “KB” and “KC”, which
+            are CloudFormation properties/key names in “A”. In the context of the
+            Hayha tool/files, “KB” and “KC” refer to their respective values, which
+            are reference pointers to either “B” or  “C”. If configured as ‘managed_protection_flow=[[“B”, “C”]]’,
```

Comment:
again, the `managed_protection_flow` is not set correctly here. it should be `[(["B"], ["C"])]`. But this section talks about `connection_to_protect`, so I think this sentence is incorrect.